lib/charms/observability_libs/v0/juju_topology.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""## Overview.
+
+This document explains how to use the JujuTopology class to
+create and consume topology information in a consistent manner.
+
+The goal of the Juju topology is to uniquely identify a piece
+of software running across any of your Juju-managed deployments.
+This is achieved by combining the following four elements:
+
+- Model name
+- Model UUID
+- Application Name
+- Unit Identifier
+
+and in some cases, the charm name.
```

Comment:
"in some cases" probably needs some clarification

Reply:
It can go altogether for now, as these are currently hypothetical scenarios.